Applications/FileIO/SWMM/SWMMInterface.cpp

```diff
@@ -0,0 +1,1248 @@
+    std::vector<GeoLib::Point*> &points, std::map<std::string,
+    std::size_t> const& name_id_map)
+{
+    std::string line;
+    getline(in, line);
+    while (!isSectionFinished(line))
+    {
+        if (isCommentLine(line))
+        {
+            getline(in, line);
+            continue;
+        }
+
+        std::vector<std::string> const split_str (BaseLib::splitString(line));
+        // Junctions = 6, Outfalls = 4, Storage = 8
+        if (split_str.size() < 4)
+        {
+            ERR ("Format not recognised.");
+            return false;
+        }
+        std::string const current_name (split_str[0]);
+        auto const it = name_id_map.find(current_name);
+        if (it == name_id_map.end())
+        {
+            ERR ("SwmmInterface::addPointElevation(): Name %s not found coordinates map.", current_name.c_str());
+            return false;
+        }
+        std::size_t const id = it->second;
+        (*points[id])[2] = BaseLib::str2number<double>(split_str[1]);
+        getline(in, line);
+    }
+    return true;
+}
+
+bool SwmmInterface::readLinksAsPolylines(std::ifstream &in,
+    std::vector<GeoLib::Polyline*> &lines, std::vector<std::string> &line_names,
+    std::vector<GeoLib::Point*> const& points, std::map<std::string, std::size_t> const& point_names)
+{
+    std::string line;
+    getline(in, line);
+    while (!isSectionFinished(line))
+    {
+        if (isCommentLine(line))
+        {
+            getline(in, line);
+            continue;
+        }
+
+        std::vector<std::string> const split_str (BaseLib::splitString(line));
+        // Conduits = 9, Pumps = 7, Weirs = 8
+        if (split_str.size() < 7)
+        {
+            ERR ("Conduit format not recognised.");
+            return false;
+        }
+
+        std::string const inlet (split_str[1]);
+        auto const i_it = point_names.find(inlet);
+        if (i_it == point_names.end())
+        {
+            ERR ("SwmmInterface::readLineElements(): Inlet node %s not found coordinates map.", inlet.c_str());
+            return false;
+        }
+
+        std::string const outlet (split_str[2]);
+        auto const o_it = point_names.find(outlet);
+        if (o_it == point_names.end())
+        {
+            ERR ("SwmmInterface::readLineElements(): Outlet node %s not found coordinates map.", outlet.c_str());
+            return false;
+        }
+        GeoLib::Polyline* ply = new GeoLib::Polyline(points);
+        std::size_t a (i_it->second);
+        ply->addPoint(i_it->second);
+        ply->addPoint(o_it->second);
+        lines.push_back(ply);
+        line_names.push_back(split_str[0]);
+        getline(in, line);
+    }
+    return true;
+}
+
+/// Deletes the geometric objects and returns false
+bool geometryCleanup(std::vector<GeoLib::Point*> &points, std::vector<GeoLib::Polyline*> &lines)
+{
+    for (auto line : lines)
+        delete line;
+    for (auto point : points)
+        delete point;
+    return false;
```

Comment:
Because it equals the return value of the geometry conversion method. It's just a short cut, but if I did this in the geo-conversion I'd have to add "return false;" at least 10 times. Now I just write "return geometryCleanup()". Since the whole method is basically just a macro to save repeating code I figured it wouldn't matter.